### ISSUE TITLE
Twitter STREAM error loops on authorization errors

### ISSUE THREAD

**Issue by justinlittman** [gwu-libraries/sfm-ui#374]
Steps to reproduce:
1. Create a bad set of Twitter credentials.
2. Use those credentials for a Twitter filter collection.
3. Start that collection.

Expected result:
Harvest terminates with an error.

Actual result:
Harvester keeps trying.

```
2016-07-15 07:32:15,663: requests.packages.urllib3.connectionpool --> Starting new HTTPS connection (27): stream.twitter.com
2016-07-15 07:32:15,775: root --> 401 Client Error: Authorization Required for url: https://stream.twitter.com/1.1/statuses/filter.json
2016-07-15 07:32:15,776: root --> sleeping 135
2016-07-15 07:32:50,928: sfmutils.harvester --> Restarting stream.
2016-07-15 07:34:30,839: root --> connecting to filter stream for {'track': u'gop cleveland, cleveland convention, gop convention, #gopconvention, #cleveland, #rnc, #gop', 'follow': u'110924703, 11134252', 'stall_warning': True}
2016-07-15 07:34:30,845: requests.packages.urllib3.connectionpool --> Starting new HTTPS connection (28): stream.twitter.com
2016-07-15 07:34:30,958: root --> 401 Client Error: Authorization Required for url: https://stream.twitter.com/1.1/statuses/filter.json
2016-07-15 07:34:30,958: root --> sleeping 140
2016-07-15 07:36:51,052: root --> connecting to filter stream for {'track': u'gop cleveland, cleveland convention, gop convention, #gopconvention, #cleveland, #rnc, #gop', 'follow': u'110924703, 11134252', 'stall_warning': True}
2016-07-15 07:36:51,057: requests.packages.urllib3.connectionpool --> Starting new HTTPS connection (29): stream.twitter.com
2016-07-15 07:36:51,170: root --> 401 Client Error: Authorization Required for url: https://stream.twitter.com/1.1/statuses/filter.json
2016-07-15 07:36:51,171: root --> sleeping 145
2016-07-15 07:39:16,271: root --> connecting to filter stream for {'track': u'gop cleveland, cleveland convention, gop convention, #gopconvention, #cleveland, #rnc, #gop', 'follow': u'110924703, 11134252', 'stall_warning': True}
2016-07-15 07:39:16,276: requests.packages.urllib3.connectionpool --> Starting new HTTPS connection (30): stream.twitter.com
2016-07-15 07:39:16,400: root --> 401 Client Error: Authorization Required for url: https://stream.twitter.com/1.1/statuses/filter.json
2016-07-15 07:39:16,400: root --> sleeping 150
```

**Comment by justinlittman** [comment]
This is being fixed by #420.

**Comment by kerchner** [comment]
Fixed by #444
